Documentation updates

TC1977 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
- Fixes the new variable name (`store_cakey` -> `store_pki`) in the deploy from Ansible docs
- Updates "mission statement" paragraph to reflect the primary use of Wireguard (and its client apps)
- Clarifies the FAQ about clients connecting to each other.
- Clarifies the prompts about VPN On Demand apply only to Apple IPsec clients.
- Encourages people to set their config options and review `config.cfg` _before_ deployment.
- Clarifies that servers (mostly) can't be upgraded to take advantage of "new Algo features".
- Adds instructions to the cloud-init docs about updating user list, and adds the correct locations of install log and configs.
- Adds a FAQ about monitoring user activity.
- Documents Python version incompatibilities described in #1622 
- Adds Road warrior instructions to the "deploy to Ubuntu" instructions.
- Reorganizes `config.cfg` and (again) encourages setting options before deployment.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Fixes #1606.
Fixes question asked in #624 (and in the Gitter chat, numerous times)
Attempts to address #1538 (again)
Closes #1596 .
Nice callback to #1291 .
Addresses #1283, #1216, #1555, closes #1600 , and again numerous questions in Gitter chat
Through a time warp, closes #1610 _before_ it was raised.
Closes #1622 .
Closes #1609 .
Documents the "BetweenClients_DROP" option brought up in #1516, #821, #1171 among others.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
Only way to test if documentation works is to see if people still have questions.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] I have updated the documentation accordingly.

jackivanov commented on README.md:
```diff
@@ -21,7 +21,6 @@ Algo VPN is a set of Ansible scripts that simplify the setup of a personal Wireg
 * Does not support legacy cipher suites or protocols like L2TP, IKEv1, or RSA
 * Does not install Tor, OpenVPN, or other risky servers
 * Does not depend on the security of [TLS](https://tools.ietf.org/html/rfc7457)
-* Does not require client software on most platforms
```
This is still true for IKEv2 and Apple devices, maybe we need to rephrase this just a bit

powerlemming replied:
You should specify it's only for Apple clients and NOT available for Windows

TC1977 replied:
Ok, hope this is better. Also I fixed the spelling of [IPsec](https://blogs.technet.microsoft.com/ianhamer/2006/02/10/an-ipsec-is-an-ipsec-is-an-ipsec/) and WireGuard.

jackivanov commented on docs/deploy-from-script-or-cloud-init-to-localhost.md:
```diff
@@ -2,7 +2,7 @@
 You can use `install.sh` to prepare the environment and deploy AlgoVPN on the local Ubuntu server in one shot using cloud-init, or run the script directly on the server after it's been created. 
 
-The script doesn't configure any parameters in your cloud, so it's on your own to configure related [firewall rules](/docs/firewalls.md), a floating ip address and other resources you may need. The output of the install script (including the p12 and CA passwords) and user config files will be installed into the `/opt/algo` directory.
+The script doesn't configure any parameters in your cloud, so you're on your own to configure related [firewall rules](/docs/firewalls.md), a floating IP address and other resources you may need. The output of the install script (including the p12 and CA passwords) and user config files will be installed into the `/opt/algo` directory. If you need to update users later, change the user list in `/ ...
```
It should work in theory, but I've never tested it. Does it work?

TC1977 replied:
It works after I fixed it. Have to activate virtualenv, etc.

davidemyers commented:
Encouraging people to review `config.cfg` before deployment is a great idea, but it still says `### Advanced users only below this line ###` immediately after the `users` section. We should consider re-ordering `config.cfg` so that only the truly advanced stuff like subnet numbering and cloud provider configuration are at the bottom of the file below this warning. Probably best done as a separate PR.

TC1977 commented:
Great feedback. I'm on the road and will try to address this after the weekend.

TC1977 commented:
> Encouraging people to review `config.cfg` before deployment is a great idea, but it still says `### Advanced users only below this line ###` immediately after the `users` section. We should consider re-ordering `config.cfg` so that only the truly advanced stuff like subnet numbering and cloud provider configuration are at the bottom of the file below this warning. Probably best done as a separate PR.

@davidemyers Done! See what you think.

davidemyers commented:
I like it. Just a few suggestions:

I think `unattended_reboot` should be "above the line" so users are encouraged to review it, but it still needs to be disabled by default.

I suggest the Advanced section start with `dnscrypt_servers` and `dns_servers`, which are probably the first things an advanced user will look for.

I also suggest burying network definitions like `strongswan_network`, `wireguard_network`, and `local_service_ip` at the bottom of the file since they'll almost never be changed and this might encourage even advanced users to leave them alone.

Thanks!

TC1977 commented:
> I like it. Just a few suggestions:
> 
> I think `unattended_reboot` should be "above the line" so users are encouraged to review it, but it still needs to be disabled by default.
> 
> I suggest the Advanced section start with `dnscrypt_servers` and `dns_servers`, which are probably the first things an advanced user will look for.
> 
> I also suggest burying network definitions like `strongswan_network`, `wireguard_network`, and `local_service_ip` at the bottom of the file since they'll almost never be changed and this might encourage even advanced users to leave them alone.
> 
> Thanks!

Ok, done! I also changed `False` to `false` for the `keys_clean_all` option, and rewrote the comment.

TC1977 commented:
Any other feedback, or anything else I should do?

TC1977 commented:
Given how many times the `BetweenClients_DROP` flag has come up, including twice in the last week on [the Gitter chat](https://gitter.im/trailofbits/algo), should that be promoted to a FAQ rather than buried in the deploy to Ubuntu instructions?

TC1977 commented:
Trying to keep these changes in sync with the latest commits. Is there a part that I can drop to get the rest of the doc updates in?

jackivanov commented:
I think we're good
